### PR TITLE
[MRG] Implement Training Job API

### LINKED ISSUE #198: [Feat] Add fine-tune training API
#### Problem description

This is a feature request: Add fine-tune training APIs, so that users can submit fine-tune jobs to the system.

#### Steps to Reproduce the Problem
N.A.

#### Expected Behavior
* User can submit a fine-tune job with the specified dataset, optimizer, learning rate...
* User can check the fine-tune job status
* User can terminate and delete the on-going fine-tune job

### PR DESCRIPTION
**Pull request by YuanmingLeee**
Closes #198 

#### What has been done to verify that this works as intended?

Start the frontend
```shell script
python -m uvicorn modelci.app.main:app --host 0.0.0.0 --reload
```
And go to the doc page `/docs`.

Test with the API:
##### 1. Create a training job
`[POST] /api/exp/train` with the following example (you should substitute the `model` to a real id of a model):
```JSON
{
  "model": "6012e189d102bac1594a9cdd",
  "data_module": {
    "dataset": "string",
    "batch_size": 0,
    "num_workers": 1,
    "data_dir": "string"
  },
  "min_epochs": 1,
  "max_epochs": 1,
  "optimizer_type": "SGD",
  "optimizer_property": {
    "lr": 0.01,
    "momentum": 0.01
  },
  "lr_scheduler_type": "StepLR",
  "lr_scheduler_property": {
    "last_epoch": -1,
    "verbose": true,
    "step_size": 6,
    "gamma": 0.01
  },
  "loss_function": "CrossEntropyLoss"
}
```
This creates a training object for you and returns you an ID:
```
{
   "id": "6018343e4d54b9d6dcddff98"
}
```

##### 2. Get all training job

`[GET] /api/exp/train/`
```JSON
[
  {
    "_id": "6018343e4d54b9d6dcddff98",
    "model": "6012e189d102bac1594a9cdd",
    "data_module": {
      "dataset": "string",
      "batch_size": 0,
      "num_workers": 1,
      "data_dir": "string"
    },
    "min_epochs": 1,
    "max_epochs": 1,
    "optimizer_type": "SGD",
    "optimizer_property": {
      "lr": 0.01,
      "momentum": 0.01
    },
    "lr_scheduler_type": "StepLR",
    "lr_scheduler_property": {
      "last_epoch": -1,
      "verbose": true,
      "step_size": 6,
      "gamma": 0.01
    },
    "loss_function": "CrossEntropyLoss",
    "status": "Unknown"
  }
]
```

##### 3. Get training job with ID

`[GET] /api/exp/train/6018343e4d54b9d6dcddff98`

```JSON
{
  "_id": "6018343e4d54b9d6dcddff98",
  "model": "6012e189d102bac1594a9cdd",
  "data_module": {
    "dataset": "string",
    "batch_size": 0,
    "num_workers": 1,
    "data_dir": "string"
  },
  "min_epochs": 1,
  "max_epochs": 1,
  "optimizer_type": "SGD",
  "optimizer_property": {
    "lr": 0.01,
    "momentum": 0.01
  },
  "lr_scheduler_type": "StepLR",
  "lr_scheduler_property": {
    "last_epoch": -1,
    "verbose": true,
    "step_size": 6,
    "gamma": 0.01
  },
  "loss_function": "CrossEntropyLoss",
  "status": "Unknown"
}
```

##### 4. Delete by ID

`[DELETE] /api/exp/train/6018343e4d54b9d6dcddff98`

Return no content as expected.

##### 5. Delete All

`[DELETE] /api/exp/train/`

Return no content as expected.

#### Why is this the best possible solution? Were any other approaches considered?

N.A.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

Experimental features, may not be presented in the official release.

#### Does this change require updates to documentation?

Yes.

#### Before submitting this PR, please make sure you have:

- [x] run `python -m pytest tests/` and confirmed all checks still pass.
- [x] verified that any code or assets from external sources are properly credited.

### CODE REVIEW

**Review by univerone — APPROVED** [review]
LGTM!
